insights/parsers/__init__.py

```diff
@@ -329,6 +329,19 @@ def keyword_search(rows, **kwargs):
     Takes a list of dictionaries and finds all the dictionaries where the
     keys and values match those found in the keyword arguments.
 
+    Keys in the row data have ' ' and '-' replaced with '_', so they can
+    match the keyword argument parsing.  For example, the keyword argument
+    'fix_up_path' will match a key named 'fix-up path'.
+
+    In addition, several suffixes can be added to the key name to do partial
+    matching of values:
+
+    * '__contains' will test whether the data value contains the given
+      value.
+    * '__startswith' tests if the data value starts with the given value
+    * '__lower_value' compares the lower-case version of the data and given
+      values.
+
```

Comment:
@gavin-romig-koch - I like those ideas.

I'm borrowing from the Django QuerySet syntax, which uses the double-underscore syntax to do both match specification (e.g. name__less_than='M' or date__year=1956) and key-relation referencing (e.g. author__name='Banks, Iain M.', where the 'author' field is a foreign key into the author table, which has a 'name' field).

In general I wanted to avoid rule code having to write Python expressions, precisely because I thought the keyword searching would be a useful feature in Fava, where you can't use Python expressions to specify matching.  After all, these functions are relatively easy to implement as list comprehensions; the keyword_search functionality just generalises the search capability out of methods that are specific to the data (e.g. "all_xfs_filesystems()" as a made-up example).

We could take a callable as a separate matcher suffix, e.g.:

```
keyword_search(fs_vfstype__matchfn=lambda x: x in ('ext2', 'ext3', 'ext4'))
```

For those tests that aren't easy to encapsulate in one match.

Does that sound like a reasonable compromise?